src/file/path.rs

```diff
@@ -0,0 +1,160 @@
+use std::{fmt, path::PathBuf};
+
+/// [ssh_config(5)](https://linux.die.net/man/5/ssh_config) file is a sequence of entries, an entry is either a *keyword argument* line, or a *comment* line.
+///
+/// In contrary to man, we will be explicitly distinguishing comments and empty lines, though originally
+/// empty lines are considered comments.
+///
+/// This is an example of ssh_config file.
+///
+/// ```ssh-config
+/// # [% sshc 0.0.1 %]
+///
+/// Include dir/folder/file
+///
+/// # a simple entry
+/// Host example.com ssh.example.com
+///     Port 22
+///     User root
+///     Ciphers aes256-cbc,arcfour
+/// ```
+#[derive(PartialEq, Debug)]
+pub struct File {
+    /// Vector of ssh_config lines.
+    ///
+    /// Please, note that '\n'-terminated ssh_config file always contains empty line in the end.
+    pub lines: Vec<Line>,
+    /// Original path of ssh_config file.
+    ///
+    /// It is used to be able to restore the source of entries provided by this file.
+    pub path: Option<PathBuf>,
+}
+
+impl fmt::Display for File {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            self.lines
+                .iter()
+                .map(ToString::to_string)
+                .collect::<Vec<String>>()
+                .join("")
+        )
+    }
+}
+
+/// Line is a sequence of characters in ssh_config file followed by /n or /r/n.
+#[derive(PartialEq, Debug)]
```

Comment:
let's add details about \r\n, \r or \n used by write_ssh_config. it is ok if we use \n in Windows?